File: 0007_ReverseInteger/Solution.py
class Solution:
    def reverse(self, x: int) -> int:
        if x is 0:
            return 0
        new_x = int(str(x)[::-1].strip('-').lstrip('0')) if x >= 0 else int('-' + str(x)[::-1].strip('-').lstrip('0'))
        if new_x > 2147483647 or new_x < -2147483648:
            return 0
        return new_x

File: 0007_ReverseInteger/Solution.cpp
class Solution {
public:
    int reverse(int x) {
        if (x == -2147483648)
        {
            return 0;
        }
        try
        {
            int sign = 1;
            if (x < 0)
            {
                sign = -1;
                x *= -1;
            }
            string str = std::to_string(x);

            std::reverse(str.begin(), str.end());

            return std::stoi(str) * sign;    
        }
        catch (...)
        {
            return 0;
        }
        
    }
};
